drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-client/src/main/java/org/drools/workbench/screens/scenariosimulation/client/handlers/CommonOnMoveHandler.java

```diff
@@ -0,0 +1,150 @@
+ * limitations under the License.
+ */
+package org.drools.workbench.screens.scenariosimulation.client.handlers;
+
+import java.util.Optional;
+
+import javax.enterprise.context.Dependent;
+
+import com.ait.lienzo.client.core.types.Point2D;
+import org.drools.scenariosimulation.api.model.FactMapping;
+import org.drools.scenariosimulation.api.model.FactMappingValue;
+import org.drools.scenariosimulation.api.model.FactMappingValueStatus;
+import org.drools.scenariosimulation.api.model.Scenario;
+import org.drools.workbench.screens.scenariosimulation.client.events.SetGridCellValueEvent;
+import org.drools.workbench.screens.scenariosimulation.client.metadata.ScenarioHeaderMetaData;
+import org.drools.workbench.screens.scenariosimulation.client.popover.ErrorReportPopoverPresenter;
+import org.drools.workbench.screens.scenariosimulation.client.popover.PopoverView;
+import org.drools.workbench.screens.scenariosimulation.client.resources.i18n.ScenarioSimulationEditorConstants;
+import org.drools.workbench.screens.scenariosimulation.client.utils.ScenarioSimulationUtils;
+import org.drools.workbench.screens.scenariosimulation.client.widgets.ScenarioGridColumn;
+import org.uberfire.ext.wires.core.grids.client.model.GridColumn;
+import org.uberfire.ext.wires.core.grids.client.widget.layer.GridLayer;
+
+/**
+ * This class is meant to provide common implementations for <b>on hover</b> behavior to be used by both mouse keyboard handler
+ */
+@Dependent
+public class CommonOnMoveHandler extends AbstractScenarioSimulationGridPanelHandler {
+
+    /* This parameter must be synchronized with POPOVER_WIDTH static variable in ErrorReportPopoverView.less */
+    private static int POPOVER_WIDTH = 200;
+    private static String NULL = "null";
+
+    protected ErrorReportPopoverPresenter errorReportPopupPresenter;
+
+    protected Integer currentlyShownHeaderRowIndex = -1;
+    protected Integer currentlyShownHeaderColumnIndex = -1;
+    protected Integer currentlyShownBodyRowIndex = -1;
+    protected Integer currentlyShownBodyColumnIndex = -1;
+
+    public void handleOnMove(final int mx, final int my) {
+        manageCoordinates(mx, my);
+    }
+
+    public void hidePopover() {
+        errorReportPopupPresenter.hide();
+    }
+
+    public void setErrorReportPopupPresenter(ErrorReportPopoverPresenter errorReportPopupPresenter) {
+        this.errorReportPopupPresenter = errorReportPopupPresenter;
+    }
+
+    @Override
+    protected boolean manageGivenExpectHeaderCoordinates(ScenarioHeaderMetaData clickedScenarioHeaderMetadata, ScenarioGridColumn scenarioGridColumn, String group, Integer uiColumnIndex) {
+        return false;
+    }
+
+    @Override
+    protected boolean manageBodyCoordinates(Integer uiRowIndex, Integer uiColumnIndex) {
+        /* In this case, the mouse is out ot the GridLayer, then return false, without perform any action */
+        if (uiColumnIndex == -1 || uiRowIndex == -1) {
```

Comment:
Related to comment in `AbstractScenarioSimulationGridPanelHandler`. This `if` would change to `uiColumnIndex == null || uiRowIndex == null`

Reply:
Agree, DONE!